### ISSUE TITLE
test_colors and test_markup_style fail when run in a clean environment

### ISSUE THREAD

**Issue by Groctel** [ManimCommunity/ManimPango#110]
I have a [script](https://git.sr.ht/~groctel/pkgbuilds/) that runs tests for this package before uploading update to the AUR. The two tests mentioned in the title fail when run in this clean environment. Here's a full log of a run of the script displaying the failure: [manimpango-log.txt](https://github.com/user-attachments/files/16567268/manimpango-log.txt).

I've found issues with other packages that depend on X11 and would fail in my Wayland desktop or in the script, which doesn't support either. Could it be that, like in that case, the SVG processing routines are being leveraged on some system dependency that behaves differently depending on the environment?

Another hypothesis is that there's a version mismatch in some dependency, but I am able to run the tests successfully after updating in my host machine, so I would discard that.

Do you have any idea what could be causing this?

**Comment by naveen521kk** [comment]
I don't have any idea why this could be happening. My guess would be an old version of Cairo - maybe try updating it?

**Comment by Groctel** [comment]
The Cairo version is the same for both envs. I've done some research and `test_wrap_text` only works when the system locales are set to something UTF-8, which makes sense as the assertion error starts with ` assert '<?xml version="1.0" encoding="UTF-8"?>`. I'm figuring out the other two failures, but I may take a while as these are being exceptionally busy weeks for me.

**Comment by Groctel** [comment]
I have isolated the failure of the tests to the presence of [cantarell-fonts](https://archlinux.org/packages/extra/any/cantarell-fonts/), which is a package that doesn't pull any dependencies, has [no sonames](https://archlinux.org/packages/extra/any/cantarell-fonts/sonames/) and provides a very conservative amount of [installed files](https://archlinux.org/packages/extra/any/cantarell-fonts/files/).

Does this make any sense to you? At a glance, I don't find any appearances of `cantarell` in the repo. I see that `fontconfig` is used for some things, but the package's file doesn't look suspicious at first. If you can't make sense of this either, I'll keep investigating.

**Comment by Groctel** [comment]
After some testing, I'm pretty sure that this is related to a pacman hook that's run when a font is installed. I have tried to run the hook's command manually but I need more knowledge because they always hang :thinking: I''ve [added a font to `checkdepends`](https://aur.archlinux.org/cgit/aur.git/commit/PKGBUILD?h=python-manimpango&id=fc87de9bf9b643c4bc7d0d92578c69e44e1e4ca5) to force the hooks to run at least once and then remove it. It's not the cleanest, but it does the trick.

From my side, it's solved albeit not perfect. Feel free to close the issue :)

**Comment by naveen521kk** [comment]
If I understand correctly, installing the package [cantarell-fonts](https://archlinux.org/packages/extra/any/cantarell-fonts/) makes the test suite pass, right? Or is it the other way around?

If that's the case, I think the system didn't have any fonts installed - which is required for the test cases to pass (I could remove that if you want - but manim won't be able to render text that way though).

Could send me the SVG files generated by those test cases, it would be better to see what's happening then. You can probably use https://docs.pytest.org/en/6.2.x/tmpdir.html#the-default-base-temporary-directory and send over the directory as a zip.

**Comment by Groctel** [comment]
Yes, it's only when the font package *is* installed. It's not an issue having to add at least one font as a check dependency with a note. No need to break manim just for this xD Here's the archive with the test files, thanks for the guide!

[tests.zip](https://github.com/user-attachments/files/17170626/tests.zip)

**Comment by naveen521kk** [comment]
> It's not an issue having to add at least one font as a check dependency with a note.

It would be best if you moved it as a dependency or an optional dependency at least. Because without a font installed manim/manimpango can't render text at all (unless using the `register_font` method).

**Comment by Groctel** [comment]
I did that some months ago but forgot to update you here. I'm closing this issue now. Thanks for your help :)
